### DISCUSSION TITLE
Thank you for making this

### DISCUSSION THREAD

**Discussion by codegod100** [okibcn/wslcompact#4] [General]
This is a much needed tool

**Comment by okibcn** [comment]
You're welcome.

Yes, the WSL team should have done something about it long ago. I just shared my solution, living for a few months as a simple function in my posh's $PROFILE.

**Comment by twiny** [comment]
```
The operation completed successfully.
```
from 361 GB to 28 GB - in few minutes. Many thanks for this tool!

**Reply by polarathene** [reply]
Does your system have 32GB or more of RAM?

I ran into an issue where while low on disk space it was still enough for the compacted disk to be written to, but `wslcompact` couldn't achieve it due to not enough memory. It seemed to use memory equivalent to the compacted size.

**Reply by twiny** [reply]
yes, that is right. i m using exactly 32GB RAM.
I did not check the RAM usage but I was also doing some work on pc.
